directory-connector/src/com/untangle/app/directory_connector/ActiveDirectoryManagerImpl.java

```diff
@@ -125,19 +127,47 @@ public List<String> getDomains(){
     }
 
     /**
-     * [getAdapter description]
-     * @param  domain [description]
-     * @return        [description]
+     * Retrieves a mapping of LDAP hosts to domains from the settings of active directory adapters.
+     * @return A map where the keys are LDAP hostnames and the values are corresponding domains.
      */
-    public ActiveDirectoryLdapAdapter getAdapter(String domain){
+    public Map<String, String> getDomainMap() {
+        Map<String, String> domainMap = new HashMap<>();
+
+        for (ActiveDirectoryLdapAdapter adAdapter : this.adAdapters) {
+            if (adAdapter == null) {
+                continue;
+            }
+            if (!adAdapter.getSettings().getEnabled()) {
+                continue;
+            }
+            
+            String ldapHost = adAdapter.getSettings().getLDAPHost();
+            String domain = adAdapter.getSettings().getDomain();
+            
+            domainMap.put(ldapHost, domain);
+        }
+        
+        return domainMap;
+    }
+
+    /**
+     * Returns the adapter that matches the specified conditions.
+     * @param domain   The domain to be matched with the adapter.
+     * @param ldapHost The hostname to be matched with the adapter.
+     * @return         The ActiveDirectoryLdapAdapter that matches the conditions, or null if no match is found.
+     */
+    public ActiveDirectoryLdapAdapter getAdapter(String domain, String ldapHost){
         for(ActiveDirectoryLdapAdapter adAdapter : this.adAdapters){
```

Comment:
What if both domain and ldapHost are null, do we have to return null ?
if yes, lets do that validation before the for loop.

Reply:
We are not allowing the user to pass null value from UI for domain and host.
![Screenshot from 2024-08-22 22-37-41](https://github.com/user-attachments/assets/7445ec87-f743-4c00-bf9b-1b1df12455b7)